### DISCUSSION TITLE
mirror() can't work when V7 to v8

### DISCUSSION THREAD

**Discussion by synesios** [gdsfactory/gdsfactory#3051] [Q&A] [answered]
<img width="604" alt="Snipaste_2024-07-28_01-42-26" src="https://github.com/user-attachments/assets/319cc11d-b71b-4d17-a355-1d5a5993bc67">
it really do work when it in the V7, i do not know how to  fix it... can you help me plz

**Comment by joamatab** [comment, marked as answer]
Hi! 

You can take a look at the migration guide for [gdsfactory8](https://gdsfactory.github.io/gdsfactory/migration.html)


```python
c = gf.Component()
ref = c << gf.c.mmi1x2()
ref.dmirror(p1=gf.kdb.DPoint(0, 0), p2=gf.kdb.DPoint(0, 1))
c
```
